File: JobSearchingWebApp/Endpoints/Oglas/GetById/OglasGetByIdResponse.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace JobSearchingWebApp.Endpoints.Oglas.GetById
{
    public class OglasGetByIdResponse
    {
        public int Id { get; set; }
        public string KompanijaNaziv { get; set; }
        public string NazivPozicije { get; set; }
        public DateTime DatumObjave { get; set; }
        public double Plata { get; set; }
        public string TipPosla { get; set; }
        public DateTime RokPrijave { get; set; }
        public DateTime? DatumModificiranja { get; set; }
        public List<OglasGetByIdResponseOglasLokacija> Lokacija { get; set; }
        public List<OglasGetByIdResponseOglasIskustvo> Iskustvo {  get; set; }
        public OglasGetByIdResponseOpisOglasa? OpisOglasa { get; set; }
    }

    public class OglasGetByIdResponseOpisOglasa
    {
        public string OpisPozicije { get; set; }
        public int MinimumGodinaIskustva { get; set; }
        public int PrefiraneGodineIskstva { get; set; }
        public string Kvalifikacija { get; set; }
        public string Vjestine { get; set; }
        public string Benefiti { get; set; }
    }

    public class OglasGetByIdResponseOglasLokacija
    {
        public int Id { get; set; }
        public string Naziv { get; set; }
    }

    public class OglasGetByIdResponseOglasIskustvo
    {
        public int Id { get; set; }
        public string Naziv { get; set; }
    }

}

File: front-end/src/app/endpoints/oglas-endpoint/get-by-id/oglas-get-by-id-response.ts
import {OglasGetResponseOglasIskustvo, OglasGetResponseOglasLokacija} from "../get/oglas-get-response";

export interface OglasGetByIdResponse {
  id: number
  nazivPozicije: string
  kompanijaNaziv: string
  logo: string | null
  datumObjave: Date
  plata: string
  tipPosla: string
  rokPrijave: Date
  datumModificiranja: Date
  iskustvo: {
    $values:OglasGetByIdResponseOglasIskustvo[]
  }
  lokacija: {
    $values: OglasGetByIdResponseOglasLokacija [];
  }
  opisOglasa: OglasGetByIdResponseOpisOglasa | null
}

export interface OglasGetByIdResponseOpisOglasa  {
  opisPozicije: string
  minimumGodinaIskustva: number
  prefiraneGodineIskstva: number
  kvalifikacija: string
  vjestine: string
  benefiti: string
}

export interface OglasGetByIdResponseOglasIskustvo {
  id: number
  naziv: string
}

export interface OglasGetByIdResponseOglasLokacija {
  id: number
  naziv: string
}
